tket/src/Utils/Expression.cpp

```diff
@@ -117,6 +117,16 @@ Expr sin_halfpi_times(const Expr& e) {
   return cos_halfpi_times(SymEngine::expand(SymEngine::integer(1) - e));
 }
 
+Expr minus_times(const Expr& e) {
+  Expr e1 = -e;
+  Expr e2 = SymEngine::expand(e1);
+
+  unsigned e1_size = e1.get_basic()->dumps().size();
+  unsigned e2_size = e2.get_basic()->dumps().size();
+
+  return e2_size < e1_size ? e2 : e1;
```

Comment:
Are there cases where `e2_size > e1_size` or can we just return `e2`?

Reply:
You can definitely construct a lot of cases where not `expand`ing would keep the expressions shorter, e.g. `Rx((a + 1)^n).dagger()`.

We could however decide nonetheless to always expand, if that makes things more consistent. I personally don't think it's necessary, but I'm happy to do it if you would prefer that.

Reply:
OK, let's leave it!